### PR TITLE
Preparations and  refactoring for #3149

### PR DESCRIPTION
**Pull request by ggreif**
This is mostly mechanical. The intention is to take out some of the patch pressure (and mental load) from #3149.

### CODE REVIEW

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -5756,6 +5755,10 @@ module Serialization = MakeSerialization(BumpStream)
 
 module Stabilization = struct
 
+  let extend64 code = code ^^ G.i (Convert (Wasm.Values.I64 I64Op.ExtendUI32))
```
Comment (nomeata) [inline src/codegen/compile.ml:5758]: My style so far was to pass `code` only when the argument can’t simply be the top element of the stack. Higher order stuff is alwasy harder to understand when called. So I’d advise against it.

You maybe could add `compile_extend64` to our list of `compile_…` functions. (And maybe they need a shorter naming convention).

**Review by nomeata — APPROVED** [review]
Minor matter-of-opinion comment, but approving to unblock you

File: src/codegen/compile.ml [hunk 2]
```diff
@@ -5544,15 +5544,14 @@ module MakeSerialization (Strm : Stream) = struct
...
 
       get_data_start ^^ get_data_size ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
       G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
-      E.else_trap_with env "data buffer not filled " ^^
+      E.else_trap_with env "data buffer not filled" ^^
 
       get_refs_size ^^
-      compile_unboxed_const 0l ^^
-      G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
+      G.i (Test (Wasm.Values.I32 I32Op.Eqz)) ^^
```
Comment (nomeata) [inline src/codegen/compile.ml:5554]: Using
```
compile_eq_const 0l
```
and optimizing there or in the peep-hole-optimizier might be a bit more declarative and less line-noisy.

Comment (ggreif) [reply src/codegen/compile.ml:5554]: Yes, #3149 already does the emissions part. I will prepare a peephole patch when I have time.

Comment (ggreif) [inline src/codegen/compile.ml:5554]:
```suggestion
      compile_unboxed_const 0l ^^
```

File: src/codegen/compile.ml (hunk 1 above)
Comment (ggreif) [reply src/codegen/compile.ml:5758]: My hope is that we'll be able to get rid of these conversions when gradually more variables/data trust to 64 bits. This `extend64` is well-confined here and   will be hopefully eliminated soon in one way or the other.

File: src/codegen/compile.ml [hunk 3]
```diff
@@ -5551,7 +5551,7 @@ module MakeSerialization (Strm : Stream) = struct
       E.else_trap_with env "data buffer not filled" ^^
 
       get_refs_size ^^
-      G.i (Test (Wasm.Values.I32 I32Op.Eqz)) ^^
+      compile_unboxed_const 0l ^^
```
Comment (nomeata) [inline src/codegen/compile.ml:5554]: `compile_eq_const`, not `compile_unboxed_const`

File: src/codegen/compile.ml [hunk 4]
```diff
@@ -5544,15 +5544,14 @@ module MakeSerialization (Strm : Stream) = struct
...
-      E.else_trap_with env "reference buffer not filled " ^^
+      E.else_trap_with env "reference buffer not filled" ^^
 
       get_data_start ^^ get_data_size ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
       G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
-      E.else_trap_with env "data buffer not filled " ^^
+      E.else_trap_with env "data buffer not filled" ^^
 
       get_refs_size ^^
       compile_unboxed_const 0l ^^
```
Comment (ggreif) [inline src/codegen/compile.ml:5554]:
```suggestion
      compile_eq_const 0l ^^
```

File: src/codegen/compile.ml (hunk 3 above)
Comment (ggreif) [reply src/codegen/compile.ml]: Yeah, silly me. Copied from the wrong place. Corrected.

File: src/codegen/compile.ml (hunk 2 above)
Comment (ggreif) [reply src/codegen/compile.ml]: I left https://github.com/dfinity/motoko/issues/3183 for the peephole.

**Pull request merged**